ludwig/config_validation/checks.py

```diff
@@ -0,0 +1,260 @@
+"""Checks that are not easily covered by marshmallow JSON schema validation like parameter interdependencies.
+
+Assumes incoming configs are comprehensive (all parameters and defaults filled in), and has been schema-validated.
+"""
+
+from abc import ABC, abstractmethod
+
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.config_validation.validation import check_schema
+from ludwig.constants import (
+    AUDIO,
+    BACKEND,
+    BINARY,
+    CATEGORY,
+    COMBINER,
+    IMAGE,
+    IN_MEMORY,
+    INPUT_FEATURES,
+    MODEL_ECD,
+    MODEL_GBM,
+    MODEL_TYPE,
+    NAME,
+    NUMBER,
+    OUTPUT_FEATURES,
+    PREPROCESSING,
+    SEQUENCE,
+    SET,
+    SPLIT,
+    TEXT,
+    TRAINER,
+    TYPE,
+    VECTOR,
+)
+from ludwig.error import ConfigValidationError
+from ludwig.types import ModelConfigDict
+from ludwig.utils.metric_utils import get_feature_to_metric_names_map
+from ludwig.utils.registry import Registry
+
+# Set of all sequence feature types.
+SEQUENCE_OUTPUT_FEATURE_TYPES = {SEQUENCE, TEXT, SET, VECTOR}
```

Comment:
worth moving to `ludwig.constants`?

Reply:
I don't see any immediate use for this being used outside of this file. Happy to do that refactoring in the future when the need arises.